### DISCUSSION TITLE
Cache Issue with Wiki.js multiple instances in Kubernetes

### DISCUSSION THREAD

**Discussion by akhilbh92** [requarks/wiki#3476] [Help / Questions]
**Question**

Describe in details what you are trying to achieve and where do you need guidance.
I used the latest stable release of wikijs docker image and passed the HA_ACTIVE as true while deploying the wiki.js in om-prem k8 cluster. I started with one replica.Then after initial application setup. Scaled it to 2 replicas. But, still I can face the cache issue. 

Please let me know what is wrong in my setup? If not, is there a way to disable cache permanently in the wikijs setup? 

**Host Info (please complete the following information):**
OS: [e.g. Ubuntu 18.04.1, Docker 19.03.6]
Wiki.js version: 2.5.170
Remote Database: [e.g. postgres 11.9.0]

**ConfigMap File:**

> apiVersion: v1
data:
  DB_HOST: <MY_DB_HOST>
  DB_PORT: "5432"
  DB_USER: "postgres"
  DB_TYPE: "postgres"
  DB_NAME: "wiki"
  HA_ACTIVE: "true"
kind: ConfigMap
metadata:
  name: postgres-sbx-config

**Deployment file:**
> apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: wikijs
  name: wikijs
spec:
  replicas: 1
  selector:
    matchLabels:
      app: wikijs
  template:
    metadata:
      labels:
        app: wikijs
    spec:
      containers:
        - name: wikijs
          image: requarks/wiki:2.5
          env:
          - name: HA_ACTIVE
            valueFrom:
              configMapKeyRef:
                key: HA_ACTIVE
                name: postgres-sbx-config 
          - name: DB_HOST
            valueFrom:
              configMapKeyRef:
                key: DB_HOST
                name: postgres-sbx-config
          - name: DB_PORT
            valueFrom:
              configMapKeyRef:
                key: DB_PORT
                name: postgres-sbx-config
          - name: DB_USER
            valueFrom:
              configMapKeyRef:
                key: DB_USER
                name: postgres-sbx-config
          - name: DB_TYPE
            valueFrom:
              configMapKeyRef:
                key: DB_TYPE
                name: postgres-sbx-config  
          - name: DB_NAME
            valueFrom:
              configMapKeyRef:
                key: DB_NAME
                name: postgres-sbx-config  
          - name: DB_PASS
            valueFrom:
              secretKeyRef:
                name: postgres-sbx 
                key: password 
          ports:
          - containerPort: 3000
            protocol: TCP

**Comment by GeorgFleig** [comment]
I do experience the same problem. Wiki.js is running with 2 replicas with HA mode enabled. Log shows this on both pods:
```
[MASTER] info: High-Availability Listener initialized successfully: [ OK ]
```

But when I change the mail settings and reload the page, sometimes the old settings from the second pod are shown. It is fixed by manually restarting the pods so the cache is cleared.

Any help is appreciated. Image: `requarks/wiki:2.5.201`.

**Reply by NGPixel** [reply]
There are a few areas where changes are not propagated correctly at the moment. This will be addressed in 3.x. In the meantime, simply restarting all the instances to fetch the latest config is the way to go.

**Reply by asyncjason** [reply]
@NGPixel I just came across the same issue. Is this issue isolated to settings only? Just trying to figure out at what stage in our deployment to scale out replicas

**Comment by vuducmanh11** [comment]
Is there any update about this issue? I think it is a big problem, it is not feasible to restart the server every time I make a change to a page and release date for v3 is uncertain

**Comment by diggi404** [comment]
still experiencing this issue. any fix yet?
